packages/main/src/Toolbar.ts

```diff
@@ -0,0 +1,582 @@
+	 * Available options are:
+	 * <ul>
+	 * <li><code>End</code></li>
+	 * <li><code>Start</code></li>
+	 * </ul>
+	 * @type {sap.ui.webc.main.types.ToolbarAlign}
+	 * @public
+	 * @defaultvalue: "End"
+	 * @name sap.ui.webc.main.Toolbar.prototype.alignContent
+	 */
+	@property({ type: ToolbarAlign, defaultValue: ToolbarAlign.End })
+	alignContent!: `${ToolbarAlign}`;
+
+	/**
+	 * Calculated width of the whole toolbar.
+	 * @private
+	 * @name sap.ui.webc.main.Toolbar.prototype.width
+	 * @type {sap.ui.webc.base.types.Integer}
+	 * @defaultvalue false
+	 */
+	@property({ type: Integer })
+	width?: number;
+
+	/**
+	 * Calculated width of the toolbar content.
+	 * @private
+	 * @name sap.ui.webc.main.Toolbar.prototype.contentWidth
+	 * @type {sap.ui.webc.base.types.Integer}
+	 * @defaultvalue 0
+	 */
+	@property({ type: Integer })
+	contentWidth?: number;
+
+	/**
+	 * Notifies the toolbar if it should show the items in a reverse way if Toolbar Popover needs to be placed on "Top" position.
+	 * @private
+	 * @type {Boolean}
+	 */
+	@property({ type: Boolean })
+	reverseOverflow!: boolean;
+
+	/**
+	 * Defines the accessible ARIA name of the component.
+	 *
+	 * @type {string}
+	 * @name sap.ui.webc.main.Toolbar.prototype.accessibleName
+	 * @defaultvalue: ""
+	 * @public
+	 */
+	@property()
+	accessibleName!: string;
+
+	/**
+	 * Receives id(or many ids) of the elements that label the input.
+	 *
+	 * @type {string}
+	 * @name sap.ui.webc.main.Toolbar.prototype.accessibleNameRef
+	 * @defaultvalue ""
+	 * @public
+	 */
+	@property({ defaultValue: "" })
+	accessibleNameRef!: string;
+
+	/**
+	* Slotted Toolbar items
+	* @type {sap.ui.webc.main.IToolbarItem[]}
+	* @name sap.ui.webc.main.Toolbar.prototype.items
+	* @slot items
+	* @public
+	*/
+	@slot({ "default": true, type: HTMLElement, invalidateOnChildChange: true })
+	items!: Array<ToolbarItem>
+
+	_onResize!: ResizeObserverCallback;
+	_onInteract!: EventListener;
+	itemsToOverflow: Array<ToolbarItem> = [];
+	itemsWidth = 0;
+	popoverOpen = false;
+	itemsWidthMeasured = false;
+
+	ITEMS_WIDTH_MAP: Map<string, number> = new Map();
+
+	static get styles() {
+		const styles = getRegisteredStyles();
+		return [
+			ToolbarCss,
+			...styles,
+		];
+	}
+
+	static get staticAreaStyles() {
+		const styes = getRegisteredStaticAreaStyles();
+		return [
+			ToolbarPopoverCss,
+			...styes,
+		];
+	}
+
+	static async onDefine() {
+		Toolbar.i18nBundle = await getI18nBundle("@ui5/webcomponents");
+	}
+
+	constructor() {
+		super();
+
+		this._onResize = this.onResize.bind(this);
+		this._onInteract = (e: Event) => this.onInteract(e as CustomEvent);
+	}
+
+	/**
+	 * Read-only members
+	 */
+
+	get overflowButtonSize(): number {
+		return this.overflowButtonDOM?.getBoundingClientRect().width || 0;
+	}
+
+	get padding(): number {
+		const toolbarComputedStyle = getComputedStyle(this.getDomRef()!);
+		return calculateCSSREMValue(toolbarComputedStyle, "--_ui5-toolbar-padding-left")
```

Comment:
There is a new CSS-scoping function that must be called for all CSS vars, used in the code, see Input.ts for an example.

We changed this 1 week ago - all component-specific CSS vars are automatically prefixed by the build for the purpose of scoping, and if you use them directly with their "raw" name, they will not work. You just need to call this function for the JS file, in the .css files nothing changes for component development.